tests/ui/hash_collision.rs

```diff
@@ -0,0 +1,6 @@
+#![warn(clippy::hash_collision)]
+use std::collections::HashMap;
+
+fn main() {
+    let example = HashMap::from([(5, 1), (5, 2)]);
```

Comment:
I'd like to see more tests. For example one with a `()` key, one with a key expr containing a `return` statement, and one with a custom key type containing a bool where `eq` always returns true and `hash` is an empty function, then `HashMap::from([(Bad(true), 1), (Bad(false)), 2)])`. That would constitute an (acceptable) false negative.

Reply:
I've now written and tested all tests you recommended, but I didn't fully understand what you meant with "key expr containg a `return` statement" 
```rust
    let _ = HashMap::from([(return Ok(()), 1), (return Err(()), 2)]); // expect no lint 
```
is what I did, but everything should already be caught by `clippy::diverging_sub_expression`, unless I missed und misunderstood something.